### PR TITLE
2574: Introduce base button component

### PR DESCRIPTION
**Pull request by steffenkleinle**
### Short description

Introduce a base button component and replace all usages of `<button>` and `role='button'`.

### Proposed changes

- Introduce a new `Button` component which overrides default styling
- Replace occurrences of  `<button>` and `role='button'` with new component
- Add aria-label in a few places

### Side effects

None.

### Resolved issues

Fixes: #2574.

---

### CODE REVIEW

**Review by f1sh1918 — APPROVED** [review]
One small thing to adjust. Rest looks fine to me. Nice refactoring

File: web/src/components/base/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  cursor: ${props => (props.disabled ? 'default' : 'pointer')};
+  pointer-events: ${props => (props.disabled ? 'none' : 'default')};
+  padding: 0;
+  border: none;
+  text-align: start;
+`
+
+type ButtonProps = {
+  onClick: () => void
+  children?: ReactNode
```
[the remark this answers is not in the document]
Comment (steffenkleinle) [reply web/src/components/base/Button.tsx:15]: Okay to me, the only case where we dont pass children is in the Modal for the Overlay. Of course we could pass a empty div or fragment. What do you think?

File: translations/translations.json [hunk 2]
```diff
@@ -6667,6 +6667,8 @@
       "detailsInformation": "Informationen",
       "detailsNextPoi": "nächster",
       "detailsPreviousPoi": "vorheriger",
+      "previousPoi": "Vorheriger Ort",
```
Comment (ztefanie) [inline translations/translations.json:6670]: No need to add these new translations you can just use ${`detailsPreviousPoi` `poi`}.

File: web/src/components/Collapsible.tsx [hunk 3]
```diff
@@ -53,10 +53,9 @@ const Collapsible = ({
   return (
     <Container className={className}>
       <CollapsibleHeader
-        role='button'
+        ariaLabel={t(collapsed ? 'showMore' : 'showLess')}
```
Comment (ztefanie) [inline web/src/components/Collapsible.tsx:56]: Better use the aria-expanded, which is explicitly for collapiables: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded

File: web/src/components/Collapsible.tsx [hunk 4]
```diff
@@ -53,10 +53,9 @@ const Collapsible = ({
...
         onClick={() => setCollapsed(!collapsed)}
-        tabIndex={0}
-        onKeyUp={() => setCollapsed(!collapsed)}>
```
Comment (ztefanie) [inline web/src/components/Collapsible.tsx:59]: Why have you decided to disable the key handling?

File: web/src/components/KebabMenu.tsx [hunk 5]
```diff
@@ -95,9 +90,9 @@ const KebabMenu = ({ items, direction, show, setShow, Footer }: KebabMenuProps):
 
   return (
     <ToggleContainer>
-      <ToggleButton onClick={onClick} aria-label={t('sideBarOpenAriaLabel')}>
+      <Button onClick={onClick} ariaLabel={t('sideBarOpenAriaLabel')}>
```
Comment (ztefanie) [inline web/src/components/KebabMenu.tsx:93]: Would also make sense to use aria-expanded here:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded

File: web/src/components/base/Button.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ztefanie) [reply web/src/components/base/Button.tsx:15]: I prefer making it child prop mandatory.

File: web/src/components/base/Button.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  tabIndex?: number
+  className?: string
+}
+
+const Button = ({ onClick, children, ariaLabel, tabIndex, className, disabled = false }: ButtonProps): ReactElement => (
+  <StyledButton
+    type='button'
+    onClick={onClick}
+    disabled={disabled}
+    aria-label={ariaLabel}
```
Comment (ztefanie) [inline web/src/components/base/Button.tsx:27]: Please also add aria role button:
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

**Review by ztefanie — COMMENTED** [review]
Tested on chrome, looks good.

File: translations/translations.json (hunk 2 above)
Comment (steffenkleinle) [reply translations/translations.json:6670]: Not the case for every language, in some languages this (i.e. the order) could differ.

File: web/src/components/Collapsible.tsx (hunk 4 above)
Comment (steffenkleinle) [reply web/src/components/Collapsible.tsx:59]: The key handling is still there, just now with the default button behavior that only specific keys (e.g. enter) trigger the onClick. This also fixes e.g. this bug: #2572.

File: web/src/components/base/Button.tsx (hunk 6 above)
Comment (steffenkleinle) [reply web/src/components/base/Button.tsx:27]: Why? This doesn't make sense for a `<button>` tag.
> The button role is for clickable elements that trigger a response when activated by the user. Adding role="button" tells the screen reader the element is a button, but provides no button functionality. Use [<button>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button) or [<input>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input) with type="button" instead.

File: web/src/components/Collapsible.tsx (hunk 3 above)
Comment (steffenkleinle) [reply web/src/components/Collapsible.tsx:56]: Good suggestion. However, I think the `ariaLabel` should still not be removed. In the example both is used together. Will add the `aria-expanded` though.

File: web/src/components/KebabMenu.tsx (hunk 5 above)
Comment (steffenkleinle) [reply web/src/components/KebabMenu.tsx:93]: Same as above.

File: web/src/components/Collapsible.tsx (hunk 3 above)
Comment (ztefanie) [reply web/src/components/Collapsible.tsx:56]: I agree

**Review by ztefanie — APPROVED** [review]
Not tested again

**Pull request merged**